tests/MongoDB.Bson.TestHelpers/TemporaryGuidRepresentationMode.cs

```diff
@@ -0,0 +1,47 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+
+namespace MongoDB.Bson.TestHelpers
+{
+    public static class TemporaryGuidRepresentationMode
```

Comment:
This class is helpful when writing tests that want to temporarily set the mode to V3 and ensure that it gets restored to the previous value at the end of the test.